### ISSUE TITLE
Profiler crashes apps on Windows 7

### ISSUE THREAD

**Issue by salsolatragus** [cqse/teamscale-profiler-dotnet#125]
When I start an app on Windows 7 with the profiler attached, app startup fails with `CLR error: 80004005. The program will now terminate`. The profiler is loaded into the runtime correctly, but at somepoint it crashes and takes the runtime down with it. The error seems to happen at different places... one we caught was:

```
KERNELBASE.dll!000007fefcc21b54()	Unknown
kernel32.dll!0000000076c91ce6()	Unknown
Profiler64.dll!Debug::logInternal(std::basic_string<char,std::char_traits<char>,std::allocator<char> > message) Line 40	C++
Profiler64.dll!Debug::log(std::basic_string<char,std::char_traits<char>,std::allocator<char> > message) Line 15	C++
Profiler64.dll!Debug::logStacktrace(std::basic_string<char,std::char_traits<char>,std::allocator<char> > context) Line 44	C++
Profiler64.dll!CProfilerCallback::handleException(std::basic_string<char,std::char_traits<char>,std::allocator<char> > context) Line 302	C++
Profiler64.dll!`CProfilerCallback::Shutdown'::`1'::catch$1() Line 157	C++
[External Code]	
Profiler64.dll!CProfilerCallback::Shutdown() Line 154	C++
Profiler64.dll!DllMain(HINSTANCE__ * hInstance, unsigned long dwReason, void * lpReserved) Line 23	C++
[External Code]
```

The error that brings us down is an [access violation](https://stackoverflow.com/questions/17168982/exception-error-c0000005-in-vc). However, this is not the original error (see `handleException()` on the stack. We couldn't capture the original error yet.

**Comment by salsolatragus** [comment]
@karottenreibe @mpdeimos Any input on this would be highly appreciated... I'll try to debug this tomorrow...

**Comment by salsolatragus** [comment]
I traced the error back to commit a0fcf3faad6104f73f444ce7cf625fa841ecedb3. It seems the debug logging somehow crashes. I tried to disable the debug logging on HEAD (c6ab14c4e35e4a267e820cd495b6d024a07e56f8) and now a can run on Win7 and get trace files with coverage.

**Comment by karottenreibe** [comment]
That debug should not be in the code anyways… :-(
Would be good to know though why this crashes on Win 7? Any idea? @salsolatragus

**Comment by salsolatragus** [comment]
It's already fixed, see PR #126 immediately above your comment ;) Feel free to take over the (small) review from Martin. [The cause](https://github.com/cqse/teamscale-profiler-dotnet/pull/126/commits/fa6aafc352f8f8b78902ea8e85b9551dea54aa65) was trivial, yet a nightmare to debug...

**Comment by salsolatragus** [comment]
Fix merged with #131 .
